experiments/download_landsat_GEE.py

```diff
@@ -0,0 +1,610 @@
+            grid_dict[gridIndex].add(new_coord)
+       
+    except OverlapError:
+        patches_s2c, center_coord = get_random_patches_grid(idx, collections, bands, crops, sampler, dates, radius, debug, grid_dict=grid_dict)
+       
+    ## random +- 30 days of random days within 1 year from the reference dates
+    #fix_random_seeds(idx)
+    #delta = timedelta(days=np.random.randint(365))
+    periods = [get_period(date, days=60) for date in dates]
+
+    collection_s2c = collections['TOA']
+    bands_s2c = bands['TOA']
+    crop_s2c = crops['TOA']
+
+    try:
+        
+        filtered_collections_s2c = [filter_collection(collection_s2c, coords, p) for p in periods]
+        patches_s2c = [get_patch(c, coords, radius, bands=bands_s2c, crop=crop_s2c) for c in filtered_collections_s2c]               
+        center_coord = coords
+
+    except (ee.EEException, urllib3.exceptions.HTTPError) as e:
+        if debug:
+            print(e)
+        grid_dict[gridIndex].add(new_coord)
+        patches_s2c, center_coord = get_random_patches_grid(idx, collections, bands, crops, sampler, dates, radius, debug, grid_dict=grid_dict)
+
+    return patches_s2c, center_coord
+
+''' sample new coord, check overlap, and get data --- match from existing coords (option3) '''
+def get_random_patches_match(idx, collections, bands, crops, sampler, dates, radius, debug=False, match_coords={}):
+    ## (lon,lat) of idx patch
+    coords = match_coords[str(idx)]
+        
+    ## random +- 30 days of random days within 1 year from the reference dates
+    #fix_random_seeds(idx)
+    #delta = timedelta(days=np.random.randint(365))
+    periods = [get_period(date, days=60) for date in dates]
+
+    collection_s2c = collections['TOA']
+    bands_s2c = bands['TOA']
+    crop_s2c = crops['TOA']
+
+    try:
+        filtered_collections_s2c = [filter_collection(collection_s2c, coords, p) for p in periods]
+        patches_s2c = [get_patch(c, coords, radius, bands=bands_s2c, crop=crop_s2c) for c in filtered_collections_s2c]               
+
+    except (ee.EEException, urllib3.exceptions.HTTPError) as e:
+        if debug:
+            print(e)
+        #patches_s2c, center_coord = get_random_patches_match(idx, collections, bands, crops, sampler, dates, radius, debug, ext_coords, rtree_obj)
+        return None, coords
+
+    return patches_s2c, coords
+
+
+''' function to save geotiff '''
+def save_geotiff(img, coords, filename):
+    #pdb.set_trace()
+    height, width, channels = img.shape
+    xres = (coords[1][0] - coords[0][0]) / width
+    yres = (coords[0][1] - coords[1][1]) / height
+    transform = Affine.translation(coords[0][0] - xres / 2, coords[0][1] + yres / 2) * Affine.scale(xres, -yres)
+    profile = {
+        'driver': 'GTiff',
+        'width': width,
+        'height': height,
+        'count': channels,
+        'crs': '+proj=latlong',
+        'transform': transform,
+        'dtype': img.dtype,
+        'compress': 'None'

```

Comment:
why not compress='lzw', predictor=2?

Reply:
Would like to know which loads faster, a smaller file that has to be decompressed, or a larger file that is already decompressed.

Reply:
No idea about that. In the end, we converted all the raw data to a single lmdb file for faster data loading.